src/devices/Seesaw/SeesawGpio.cs

```diff
@@ -0,0 +1,150 @@
+        public void WriteGpioDigital(byte pin, bool value)
+        {
+            if (pin < 0 || pin > 63)
+            {
+                throw new ArgumentException("Gpio pin must be within 0-63 range.");
+            }
+
+            WriteGpioDigitalBulk((UInt64)(1 << pin), value);
+        }
+
+        /// <summary>
+        /// Write a value to a number of GPIO pins
+        /// </summary>
+        /// <param name="pins">A 64bit integer containing 1 bit for each pin. If a bit is set to 1 then the pin value is set for the associated pin.</param>
+        /// <param name="value">The pin value to be set.</param>
+        public void WriteGpioDigitalBulk(UInt64 pins, bool value)
+        {
+            if (!HasModule(Seesaw_Module.Gpio))
+            {
+                throw new NotSupportedException($"The hardware on I2C Bus {I2cDevice.ConnectionSettings.BusId}, Address 0x{I2cDevice.ConnectionSettings.DeviceAddress:X2} does not support Adafruit SeeSaw GPIO functionality");
+            }
+
+            byte[] buffer = new byte[] { (byte)(pins >> 24), (byte)(pins >> 16), (byte)(pins >> 8), (byte)pins, (byte)(pins >> 56), (byte)(pins >> 48), (byte)(pins >> 40), (byte)(pins >> 32) };
+
+            Write(Seesaw_Module.Gpio, value ? Seesaw_Function.Gpio_Bulk_Set : Seesaw_Function.Gpio_Bulk_Clr, buffer);
+        }
+
+        /// <summary>
+        /// Read a value from a GPIO pin.
+        /// </summary>
+        /// <param name="pin">The pin that has its value read.</param>
+        /// <returns>A boolean value representaing the status of the GPIO pin.</returns>
+        public bool ReadGpioDigital(byte pin)
+        {
+            if (pin < 0 || pin > 63)
+            {
+                throw new ArgumentException("Gpio pin must be within 0-63 range.");
+            }
+
+            return((ReadGpioDigitalBulk((UInt64)(1 << pin)) & (UInt64)(1 << pin)) != 0);
+        }
+
+        /// <summary>
+        /// Read a value from a number of GPIO pins.
+        /// </summary>
+        /// <param name="pins">A 64bit integer containing 1 bit for each pin. If a bit is set to 1 then the pin value is read for the associated pin.</param>
+        /// <returns>A 64bit integer containing 1 bit for each pin status requested.</returns>
+        public UInt64 ReadGpioDigitalBulk(UInt64 pins)
+        {
+            if (!HasModule(Seesaw_Module.Gpio))
+            {
+                throw new NotSupportedException($"The hardware on I2C Bus {I2cDevice.ConnectionSettings.BusId}, Address 0x{I2cDevice.ConnectionSettings.DeviceAddress:X2} does not support Adafruit SeeSaw GPIO functionality");
+            }
+
+            byte[] buffer = Read(Seesaw_Module.Gpio, Seesaw_Function.Gpio_Bulk, 8);
+
+            return ((((UInt64)buffer[4] << 56) | ((UInt64)buffer[5] << 48) | ((UInt64)buffer[6] << 40) | ((UInt64)buffer[7] << 32) | ((UInt64)buffer[0] << 24) | ((UInt64)buffer[1] << 16) | ((UInt64)buffer[2] << 8) | buffer[3]) & pins);
```

Comment:
https://docs.microsoft.com/en-us/dotnet/api/system.buffers.binary.binaryprimitives?view=netstandard-2.1

Reply:
@krwq Unfortunately the byte array is not formatted as a ulong converted ty bytes but as two uints representing the low and high 32 bits and so there is no obvious bit conversion that I can see.
I have created functions for this conversion and documented the byte array layout and it will be available in the next update